python/mxnet/optimizer.py

```diff
@@ -40,10 +40,39 @@ class Optimizer(object):
 
     begin_num_update : int, optional
         The initial number of updates.
+
+    do_pruning : boolean, optional
```

Comment:
write your own optimizer that subclass SGD instead of putting this in the base optimizer class